tiledb/api/src/stats.rs

```diff
@@ -0,0 +1,70 @@
+use crate::error::Error;
+use crate::Result as TileDBResult;
+
+pub fn stats_enable() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_enable()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+       Ok(())
+    } else {
+        Err(Error::from("TileDB stats were not successfully enabled."))
+    }
+}
+
+pub fn stats_disable() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_disable()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+        Ok(())
+    } else {
+        Err(Error::from("TileDB stats were not successfully disabled."))
+    }
+}
+
+pub fn stats_reset() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_reset()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+       Ok(())
+    } else {
+       Err(Error::from("TileDB stats were not successfully reset."))
+    }
+}
+
+pub fn stats_dump_str() -> TileDBResult<String> {
+    let mut stats_dump_c_str = std::ptr::null_mut::<std::os::raw::c_char>();
+    let res = unsafe {
+        ffi::tiledb_stats_dump_str(
```

Comment:
Is it safe to call this when stats are not enabled?

Reply:
The call is safe as long as we properly account for whether it returns nullptr or an empty string. Though I was skimming the implementation and we should really check whether there are corner cases on that.